internal/graph/checkingresourcestream.go

```diff
@@ -0,0 +1,537 @@
+				if err != nil {
+					crs.setError(err)
+					return err
+				}
+			}
+
+			crs.rq.markResourceCompleted(current)
+		}
+	}
+}
+
+// setError sets an error that occurred.
+func (crs *checkingResourceStream) setError(err error) {
+	crs.errSetter.Do(func() {
+		crs.err = err
+		crs.cancel()
+	})
+}
+
+// process is a processing worker for a reachable resources result, performing checks if necessary.
+func (crs *checkingResourceStream) process() {
+	defer crs.processingWaitGroup.Done()
+
+	for {
+		select {
+		case <-crs.ctx.Done():
+			return
+
+		case <-crs.reachableResourcesCompleted:
+			for {
+				ok, err := crs.runProcess(true)
+				if err != nil {
+					crs.setError(err)
+					return
+				}
+				if !ok {
+					break
+				}
+			}
+			return
+
+		case <-crs.availableForProcessing:
+			for {
+				ok, err := crs.runProcess(false)
+				if err != nil {
+					crs.setError(err)
+					return
+				}
+				if !ok {
+					break
+				}
+			}
+			continue
+		}
+	}
+}
+
+func (crs *checkingResourceStream) runProcess(alwaysProcess bool) (bool, error) {
+	// Collect any resources that need to be checked, up to the configured limit, and issue a check.
+	// If a resource does not require a check, simply place on the toPublish queue.
+	toCheck := util.NewMultiMap[string, possibleResource]()
+	toProcess := crs.rq.selectResourcesToProcess(alwaysProcess)
+	if len(toProcess) == 0 {
+		return false, nil
+	}
+
+	for _, current := range toProcess {
+		if current.reachableResult.Resource.ResultStatus == v1.ReachableResource_HAS_PERMISSION {
+			current.lookupResult = &v1.DispatchLookupResourcesResponse{
+				ResolvedResource: &v1.ResolvedResource{
+					ResourceId:     current.reachableResult.Resource.ResourceId,
+					Permissionship: v1.ResolvedResource_HAS_PERMISSION,
+				},
+				Metadata:            addCallToResponseMetadata(current.reachableResult.Metadata),
+				AfterResponseCursor: current.reachableResult.AfterResponseCursor,
+			}
+			crs.rq.updateToBePublished(current)
+			continue
+		}
+
+		toCheck.Add(current.reachableResult.Resource.ResourceId, current)
+	}
+
+	if toCheck.IsEmpty() {
+		crs.availableForPublishing <- true
+		return true, nil
+	}
+
+	// Issue the bulk check over all the resources.
+	results, resultsMetas, err := computed.ComputeBulkCheck(
+		crs.ctx,
+		crs.checker,
+		computed.CheckParameters{
+			ResourceType:  crs.req.ObjectRelation,
+			Subject:       crs.req.Subject,
+			CaveatContext: crs.req.Context.AsMap(),
+			AtRevision:    crs.req.Revision,
+			MaximumDepth:  crs.req.Metadata.DepthRemaining,
+			DebugOption:   computed.NoDebugging,
+		},
+		toCheck.Keys(),
+	)
+	if err != nil {
+		return true, err
+	}
+
+	for _, rai := range toCheck.Values() {
+		checkResult := results[rai.reachableResult.Resource.ResourceId]
+
+		var permissionship v1.ResolvedResource_Permissionship
+		var missingFields []string
+
+		switch {
+		case checkResult == nil || checkResult.Membership == v1.ResourceCheckResult_NOT_MEMBER:
```

Comment:
this case is not being tested by the newly added test in `internal/dispatch/graph/lookupresources_test.go`